src/database/mod.rs

```diff
@@ -158,6 +170,17 @@ pub trait Database: BatchOperations {
     ///
     /// It should insert and return `0` if not present in the database
     fn increment_last_index(&mut self, keychain: KeychainKind) -> Result<u32, Error>;
+
+    /// Delete UTXOs provided as params, if `spent_utxos` is empty it deletes all spent
+    /// UTXOs in the database.
+    fn del_spent_utxos(&mut self, spent_utxos: Vec<LocalUtxo>) -> Result<Vec<LocalUtxo>, Error>;
```

Comment:
I feel the API would be more clear if we asked for `Option<Vec<OutPoint>>` .. `LocalUtxo` is an internal structure we use to store more metadata about an utxo.. But an utxo could be identified by an `OutPoint`. And we should not burden users to provide a `LocalUtxo`. We should be able to fetch a `LocalUtxo` given an `OutPoint`. 

And delete all utxos if user provide `None` instead of a list of outpoints..

Reply:
This makes total sense, I will make this change.

Reply:
Yes it is.. Along with all other bdk specific types.. They are useful as a holder of extra meta data around common structures, this one around an outpoint.. So they makes sense as function outputs.. But not as inputs.. Because then you are forcing users to find out extra data that they don't really need for the API call..